src/joystick/hidapi/SDL_hidapi_sinput.c

```diff
@@ -0,0 +1,745 @@
+{
+    return SDL_Unsupported();
+}
+
+static Uint32 HIDAPI_DriverSInput_GetJoystickCapabilities(SDL_HIDAPI_Device *device, SDL_Joystick *joystick)
+{
+    Uint32 caps = 0;
+    caps |= SDL_JOYSTICK_CAP_RUMBLE;
+    caps |= SDL_JOYSTICK_CAP_PLAYER_LED;
+    return caps;
+}
+
+static bool HIDAPI_DriverSInput_SetJoystickLED(SDL_HIDAPI_Device *device, SDL_Joystick *joystick, Uint8 red, Uint8 green, Uint8 blue)
+{
+    return SDL_Unsupported();
+}
+
+static bool HIDAPI_DriverSInput_SendJoystickEffect(SDL_HIDAPI_Device *device, SDL_Joystick *joystick, const void *data, int size)
+{
+    return SDL_Unsupported();
+}
+
+static bool HIDAPI_DriverSInput_SetJoystickSensorsEnabled(SDL_HIDAPI_Device *device, SDL_Joystick *joystick, bool enabled)
+{
+    return true;
+}
+
+static void HIDAPI_DriverSInput_HandleStatePacket(SDL_Joystick *joystick, SDL_DriverSInput_Context *ctx, Uint8 *data, int size)
+{
+    Sint16 axis = 0;
+    Sint16 accel = 0;
+    Sint16 gyro = 0;
+    Uint64 timestamp = SDL_GetTicksNS();
+    float imu_values[3] = { 0 };
+
+    if (ctx->last_state[SINPUT_REPORT_IDX_BUTTONS_0] != data[SINPUT_REPORT_IDX_BUTTONS_0]) {
+        SDL_SendJoystickButton(timestamp, joystick, SDL_GAMEPAD_BUTTON_SOUTH, (data[SINPUT_REPORT_IDX_BUTTONS_0] & 0x01));
+        SDL_SendJoystickButton(timestamp, joystick, SDL_GAMEPAD_BUTTON_EAST,  (data[SINPUT_REPORT_IDX_BUTTONS_0] & 0x02));
+        SDL_SendJoystickButton(timestamp, joystick, SDL_GAMEPAD_BUTTON_WEST,  (data[SINPUT_REPORT_IDX_BUTTONS_0] & 0x04));
+        SDL_SendJoystickButton(timestamp, joystick, SDL_GAMEPAD_BUTTON_NORTH, (data[SINPUT_REPORT_IDX_BUTTONS_0] & 0x08));
+
+        Uint8 hat;
+        switch ((data[SINPUT_REPORT_IDX_BUTTONS_0] & 0xF0) >> 4) {
+        case 0b0001:
+            hat = SDL_HAT_UP;
+            break;
+        case 0b1001:
+            hat = SDL_HAT_RIGHTUP;
+            break;
+        case 0b1000:
+            hat = SDL_HAT_RIGHT;
+            break;
+        case 0b1010:
+            hat = SDL_HAT_RIGHTDOWN;
+            break;
+        case 0b0010:
+            hat = SDL_HAT_DOWN;
+            break;
+        case 0b0110:
+            hat = SDL_HAT_LEFTDOWN;
+            break;
+        case 0b0100:
+            hat = SDL_HAT_LEFT;
+            break;
+        case 0b0101:
+            hat = SDL_HAT_LEFTUP;
+            break;
+        default:
+            hat = SDL_HAT_CENTERED;
+            break;
+        }
+        SDL_SendJoystickHat(timestamp, joystick, 0, hat);
+    }
+
+    
+    if (ctx->last_state[SINPUT_REPORT_IDX_BUTTONS_1] != data[SINPUT_REPORT_IDX_BUTTONS_1]) {
+        SDL_SendJoystickButton(timestamp, joystick, SDL_GAMEPAD_BUTTON_LEFT_STICK,  (data[SINPUT_REPORT_IDX_BUTTONS_1] & 0x01));
+        SDL_SendJoystickButton(timestamp, joystick, SDL_GAMEPAD_BUTTON_RIGHT_STICK, (data[SINPUT_REPORT_IDX_BUTTONS_1] & 0x02));
```

Comment:
These should be boolean values, e.g. `(data[SINPUT_REPORT_IDX_BUTTONS_1] & 0x02) != 0`